### PR TITLE
Compression choice

### PR DESCRIPTION
**Pull request by bassem97**

### CODE REVIEW

File: src/UpupUploader.tsx [hunk 1]
```diff
@@ -13,18 +13,32 @@ interface UpupUploaderProps {
...
+ * @constructor
+ */
 export const UpupUploader: FC<UpupUploaderProps> = ({
                                                         client,
                                                         bucket,
                                                         setKey,
                                                         canUpload,
                                                         provider,
+                                                        compressible = false,
                                                         API_KEY,
```
Comment (SaddemAmine) [inline src/UpupUploader.tsx:42]: I suggest we pass these in a single config object to avoid overly crowded props

File: src/UpupUploader.tsx [hunk 2]
```diff
@@ -13,18 +13,32 @@ interface UpupUploaderProps {
...
+ * @param CLIENT_ID the OAuth client ID
+ * @param compressible whether the user want to compress the file before uploading it or not. Default value is false
+ * @constructor
+ */
 export const UpupUploader: FC<UpupUploaderProps> = ({
                                                         client,
                                                         bucket,
                                                         setKey,
                                                         canUpload,
                                                         provider,
```
Comment (SaddemAmine) [inline src/UpupUploader.tsx:40]: Maybe pass the config object I mentioned in the other comment and the provider in the same prop

File: src/UpupUploader.tsx [hunk 3]
```diff
@@ -36,6 +50,7 @@ export const UpupUploader: FC<UpupUploaderProps> = ({
             bucket={bucket}
             setKey={setKey}
             canUpload={canUpload}
+            compressible={compressible}
```
Comment (SaddemAmine) [inline src/UpupUploader.tsx]: Maybe `compress` makes more sense than `compressible` ?

File: src/components/UploadFiles.tsx [hunk 4]
```diff
@@ -16,9 +17,16 @@ export interface UploadFilesProps {
  * @param bucket bucket name
  * @param setKey return the final name of the file, usually it has timestamp prefix
  * @param canUpload to control when to upload the file , it has default false value
+ * @param compressible whether the user want to compress the file before uploading it or not. Default value is false
  * @constructor
  */
-export const UploadFiles: FC<UploadFilesProps>  = ({client,bucket,setKey, canUpload}: UploadFilesProps) => {
+export const UploadFiles: FC<UploadFilesProps> = ({
```
Comment (SaddemAmine) [inline src/components/UploadFiles.tsx:23]: prettier 🥴

File: src/components/UploadFiles.tsx [hunk 5]
```diff
@@ -27,29 +35,28 @@ export const UploadFiles: FC<UploadFilesProps>  = ({client,bucket,setKey, canUpl
...
+        else
+            filesToUpload = files;
 
-                compressedFiles.push(compressedFile);
-            }
-        }
+        if (filesToUpload) {
+            filesToUpload.forEach(fileToUpload => {
+                // assign a unique name for the file, usually has to timestamp prefix
+                key = `${Date.now()}__${fileToUpload.name}`
```
Comment (SaddemAmine) [inline src/components/UploadFiles.tsx:51]: Shouldn't this be a default value if the `key` prop is undefined ?

Comment (SaddemAmine) [reply src/components/UploadFiles.tsx:51]: if so, we are not using said prop

File: src/components/googleDrive.tsx [hunk 6]
```diff
@@ -85,14 +88,22 @@ export const GoogleDrive: FC<GoogleDriveProps> = ({client,bucket, API_KEY, APP_I
...
+                fileToUpload = await response.arrayBuffer().then((buffer) =>
+                    new File([buffer], document[window.google.picker.Document.NAME])
+                )
+
+
+            console.log(fileToUpload)
 
             // assign a unique name for the file, usually has to timestamp prefix
-            const key = `${Date.now()}__${compressedFile.name}`
+            const key = `${Date.now()}__${fileToUpload.name}`
```
Comment (SaddemAmine) [inline src/components/googleDrive.tsx:103]: Same as previous key comment

**Review by SaddemAmine — CHANGES_REQUESTED** [review]
Some comments have multiple instances but I only pointed them out once. Feel free to let me know if that requires extra clarification.

File: src/UpupUploader.tsx (hunk 3 above)
Comment (bassem97) [reply src/UpupUploader.tsx]: the best practices says that the  name of boolean has a prifix with: is/has/can + noun like isCompressed , hasAge, canUpload OR simply the adjective like sorted, hidden , compressible

File: src/components/UploadFiles.tsx (hunk 5 above)
Comment (bassem97) [reply src/components/UploadFiles.tsx:51]: the value of the key comes from fileToUpload and if its undefined we can't make all the upload process

File: src/UpupUploader.tsx (hunk 1 above)
Comment (bassem97) [reply src/UpupUploader.tsx:42]: yes , true

File: src/UpupUploader.tsx (hunk 3 above)
Comment (bassem97) [reply src/UpupUploader.tsx]: i will change it to shouldBeCompressed on the next release

Comment (SaddemAmine) [reply src/UpupUploader.tsx]: We settled on `toBeCompressed`

File: src/UpupUploader.tsx (hunk 2 above)
Comment (SaddemAmine) [reply src/UpupUploader.tsx:40]: @bassem97 will be tacking this in the next patch

**Review by SaddemAmine — APPROVED** [review]

File: src/UpupUploader.tsx (hunk 1 above)
Comment (bassem97) [reply src/UpupUploader.tsx:42]: linked an issue for that https://github.com/uNotesOfficial/upup/issues/6

**Review by jondav1 — APPROVED** [review]

**Review by AminDhouib — APPROVED** [review]

**Pull request merged**